### PR TITLE
Training template

### PR DESCRIPTION
**Pull request by BrianPugh**
**Pull request recommendations:**
- [ ] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [ ] Link to any relevant issue in the PR description. Ex: _Resolves [gh-12], adds tiff file format support_
- [ ] Provide context of changes.
- [ ] Provide relevant tests for your feature or bug fix.
- [ ] Provide or update documentation for any feature added by your pull request.

Thanks for contributing!

### CODE REVIEW

File: experiments/train.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+experiment_path = this_file_dir / experiment_name
+if not experiment_path.is_dir():
+    raise FileNotFoundError(f"No such directory {experiment_path}")
+
+sys.path.insert(0, str(experiment_path))
+
+# We switch the working directory so that hydra's automatic path handling
+# happens within the experiment directory.
+with working_dir(experiment_path):
```
Comment (omry) [inline experiments/train.py:81]: you can just customize the Hydra working directory directly.
see [this](https://hydra.cc/docs/configure_hydra/workdir).

Comment (BrianPugh) [reply experiments/train.py:81]: now correct me if i'm wrong, but that would require me changing the configuration in each experiment folder. Here I am sharing a single unified `train.py` driver to handle all my hydra and pytorch setup.

Comment (omry) [reply experiments/train.py:81]: Can you point to something specific to understand what you mean?

Generally speaking, you can (and should) reuse your configuration and compose what you need.
specifically, you can include the same hydra config snippet from all your experiments.

base_config.yaml
```
hydra:
  run:
    dir: ...
```

experiment.yaml
```yaml
defaults:
 - base_config
 - other: defaults

other: stuff
```

Comment (BrianPugh) [reply experiments/train.py:81]: alright, i get what you're saying, but for my use case thats actually what i want to do.

1. Each experiments code is unique, the template is just some boilerplate/example code to get you going.
2. I want each experiment to be bundled with its default configuration so that each experiment is mostly isolated. This way the only backwards compatible elements between experiments are located in `train.py` and my package `pugh_torch`
3. I get that Hydra's way would be to shove all of that into a single config folder at `experiment/config`, and that would be convenient for some common configurations, but now each experiment's code/config is located in two places and has less-straightforward configuration (from a perspective of someone who is new to Hydra).

Simply put, I believe the current setup is more robust to people that don't fully understand Hydra from messing up other experiments than the one they are working on. It loses the benefits you mention, but it makes code reviews easier and getting new people started easier.
